vulkan_hal.cpp

```diff
@@ -49,13 +50,20 @@ static VkResult AcquireImageANDROID(VkDevice, VkImage /*dev*/,
                                     int nativeFenceFd,
                                     VkSemaphore /*semaphore*/,
                                     VkFence /*fence*/) {
+  // wait for fence to signal before acquiring image
+  sync_wait(nativeFenceFd, -1);
+
   close(nativeFenceFd);
   return VK_SUCCESS;
 }
 
-static VkResult QueueSignalReleaseImageANDROID(VkQueue /*queue*/,
+static VkResult QueueSignalReleaseImageANDROID(VkQueue queue,
                                                VkImage /*image*/,
                                                int* pNativeFenceFd) {
+  // wait for any operations on queue to complete
+  // before releasing image
+  vkQueueWaitIdle(queue);
+
```

Comment:
The deadlock may happen if the command in queue need waiting for VkEvent, the driver needs to set it through vkResetEvent, but may be blocked by vkQueueWaitIdle here.

Also, the tearing occurs because the app may update the buffer even when it under display, we need get the display release fence in this function.